### PR TITLE
Add sample for error correction

### PR DESCRIPTION
**Pull request by ShammiL**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: errorCorrection/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+1. Create a [Sapling account](https://sapling.ai).
+2. Obtain an API key an include it in the `Config.toml` file.
+```
+apiKey = "<API_KEY>"
+```
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+bal run
```
Comment (SasinduDilshara) [inline errorCorrection/README.md]:
```suggestion
$ bal run
```

File: errorCorrection/service.bal [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    resource function post spellings(SpellCheckRequest request) returns error? {
+        SaplingRequest saplingRequest = {
+            'key: apiKey,
+            text: request.content,
+            session_id: "session1"
+        };
+        SaplingResponse response = check saplingClient->/api/v1/spellcheck.post(saplingRequest);
+        int errorCount = 0;
+        foreach EditBody edit in response.edits {
+            io:println(string `${edit.sentence} : ${edit.replacement}`);
```
Comment (SasinduDilshara) [inline errorCorrection/service.bal]:
```suggestion
            io:println(string `${edit.sentence}: ${edit.replacement}`);
```

File: errorCorrection/.devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (pcnfernando) [inline errorCorrection/.devcontainer.json]: Can remove this

File: errorCorrection/Ballerina.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "etl"
+name = "errorCorrection"
+version = "0.1.0"
+distribution = "2201.8.0"
+
+[build-options]
```
Comment (pcnfernando) [inline errorCorrection/Ballerina.toml]: Can remove this

File: errorCorrection/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+```
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+$ bal run
+```
+
+### Send a request
+POST http://localhost:8080/api/posts/spellings
```
Comment (pcnfernando) [inline errorCorrection/README.md]: shall we have the exact curl instead

File: errorCorrection/Ballerina.toml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[package]
+org = "etl"
+name = "errorCorrection"
```
Comment (pcnfernando) [inline errorCorrection/Ballerina.toml:3]: Use the same convernsion as the others

File: errorCorrection/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+## Steps to run the sample
+
+### 1. Configuring the Config.toml file
+1. Create a `Config.toml` file in the root directory and add the below content.
+```toml
+apiKey = "<API_KEY>"
+```
+2. Create a [Sapling account](https://sapling.ai) and obtain an API key and replace with `<API_KEY>`.
```
Comment (pcnfernando) [inline errorCorrection/README.md:8]:
```suggestion
2. Create a [Sapling account](https://sapling.ai) and obtain an API key and use it in place of `<API_KEY>` in `Config.toml`.
```

File: errorCorrection/service.bal [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    int 'start;
+|};
+
+type SaplingResponse record {|
+    EditBody[] edits;
+|};
+
+final http:Client saplingClient = check new ("https://api.sapling.ai");
+
+service /api/posts on new http:Listener(8080) {
```
Comment (pcnfernando) [inline errorCorrection/service.bal:31]: Make them isolated

File: errorCorrection/service.bal [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    string sentence;
+    int sentence_start;
+    int 'start;
+|};
+
+type SaplingResponse record {|
+    EditBody[] edits;
+|};
+
+final http:Client saplingClient = check new ("https://api.sapling.ai");
```
Comment (pcnfernando) [inline errorCorrection/service.bal:29]: Can be readonly

**Review by pcnfernando — APPROVED** [review]
Suggestions will be handled separately. Hence approving

**Pull request merged**